ext/LinearSolveMKLExt.jl

```diff
@@ -0,0 +1,54 @@
+module LinearSolveMKLExt
+
+using MKL_jll
+using LinearAlgebra: BlasInt, LU
+using LinearAlgebra.LAPACK: require_one_based_indexing, chkfinite, chkstride1, chkargsok
+using LinearAlgebra
+const usemkl = MKL_jll.is_available()
+
+using LinearSolve
+using LinearSolve: ArrayInterface, MKLLUFactorization, @get_cacheval, LinearCache, SciMLBase
+
+function getrf!(A::AbstractMatrix{<:Float64}; ipiv = similar(A, BlasInt, min(size(A,1),size(A,2))), info = Ref{BlasInt}(), check = false)
+    require_one_based_indexing(A)
+    check && chkfinite(A)
+    chkstride1(A)
+    m, n = size(A)
+    lda  = max(1,stride(A, 2))
+
+    if isempty(ipiv)
+        ipiv = similar(A, BlasInt, min(size(A,1),size(A,2)))
+    end
+
+    ccall((:dgetrf_, MKL_jll.libmkl_rt), Cvoid,
```

Comment:
https://github.com/JuliaLinearAlgebra/BLASBenchmarksCPU.jl/blob/master/src/ccallblas.jl

I don't seem to be adding a trailing underscore to `dgetrf_` `_`